### ISSUE TITLE
Risc-V and perhaps x86, issue with NaN

### ISSUE THREAD

**Issue by stensmo** [JuliaLang/julia#56672]
While testing [Numbers NaN test](https://github.com/JuliaLang/julia/blob/master/test/numbers.jl#L2917), some tests related to NaN are failing. The issue relates to negative NaN and positive Nan. Why is it so important that the sign is correct? In that case:

On X86 
julia> 0.0*Inf64===-NaN64
true

On Risc-V
julia> 0.0*Inf64===NaN64
true

Why does X86 provide the negative Nan? If you only look at the type and use isnan, everything works as expected on all platforms. In that case, the tests can be modified.

**Comment by oscardssmith** [comment]
I believe (but am not sure) that these tests aren't actually guaranteed to hold with LLVM's NaN semantics. @Keno can you confirm?

**Comment by giordano** [comment]
These tests were introduced by @StefanKarpinski in #23253

**Comment by oscardssmith** [comment]
yeah the issue is that I'm pretty sure that preserving nan payloads is explicitly not guaranteed by LLVM

**Comment by stensmo** [comment]
> yeah the issue is that I'm pretty sure that preserving nan payloads is explicitly not guaranteed by LLVM

Here it says that it wont,  [Behavior of Floating-Point NaN values](https://llvm.org/docs/LangRef.html#id1985)](https://llvm.org/docs/LangRef.html#floatnan)

Floating-point math operations that return a NaN are an exception from the general principle that LLVM implements IEEE-754 semantics. Unless specified otherwise, the following rules apply whenever the IEEE-754 semantics say that a NaN value is returned: *the result has a non-deterministic sign*; the quiet bit and payload are non-deterministically chosen from the following set of options:

**Comment by giordano** [comment]
Consistency issues with NaNs are discussed in more details in #49353

**Comment by stensmo** [comment]
Julia must have the exact same semantics as LLVM or as a patched version of LLVM. These tests violate the semantics of the current LLVM used, and the tests should be removed, and there are some more in the tests for complex. 

On the bright side, Julia for RISC-V passes all vital tests and is fully usable (except that no binary packages are avaiable)

**Comment by oscardssmith** [comment]
yeah I'm pretty sure that all of these tests are invalid.

**Comment by StefanKarpinski** [comment]
I'm fine with deleting these. Would be nice if payloads were actually a usable, reliable feature, but that's not the reality we live in.
